backend/src/controllers/groups-controller.ts

```diff
@@ -0,0 +1,106 @@
+import { Request, Response } from 'express';
+import Group from '../models/group';
+import { validationResult } from 'express-validator';
+import group from "../models/group";
+import { ObjectId } from "mongodb";
+
+
+
+
+
+//Creates a group with the desired name and gives a unique ID
+export const createGroup = async (req: Request, res: Response) => {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+        console.log("errors", errors);
+        return res.status(400).json({ errors: errors.array() });
+    }
+    //Checks if there's a similar group
+    const dupeGroup = group.findOne({ name: req.body })
+    if (!dupeGroup) {
+        return res.status(409).json({ error: "A group with the same name already exists."  });
+    }
+    else{
+        //Creates a new group with name and id
+        try {
+            const name = req.body.name as string;
+
+            const group = new Group({
+                name: name,
+            });
+            await group.save();
+            return res.status(201).json({group});
+        } catch (error: any) {
+            res.status(500).json({ message: 'Server error', error: error.message });
+        }
+    }
+}
+
+//Returns the existing groups
+export const retrieveGroups = async (req: Request, res: Response) => {
+    try {
+        const groups = await Group.find();
+        return res.status(200).json(groups);
+    } catch (error: any) {
+        return res.status(500).json({ message: 'Server error', error: error.message });
+    }
+}
+
+//Find specific group by ID
+export const getSpecificGroup = async (req: Request, res:Response) => {
+    const groupID: string = req.params._id;
```

Comment:
When I make a call to this route, it doesn't work yet -- see my suggestion for a fix in the `group-routes.ts` file below